### PR TITLE
DM-39139: improve base class for server tests by making it more general, and improve some documentation

### PR DESCRIPTION
**Pull request by r-owen**

### CODE REVIEW

File: python/lsst/ts/tcpip/constants.py [hunk 1]
```diff
@@ -23,17 +23,21 @@
...
 
-# localhost constants
+# localhost constants for IPV4 and IPV6
 LOCALHOST_IPV4 = "127.0.0.1"
 LOCALHOST_IPV6 = "::1"
-LOCAL_HOST = LOCALHOST_IPV4  # Deprecated
+# The default localhost
+LOCALHOST = LOCALHOST_IPV4
+# Deprecated alias to LOCALHOST
+LOCAL_HOST = LOCALHOST
```
Comment (wvreeven) [inline python/lsst/ts/tcpip/constants.py:40]: Out of curiosity, when do you plan to remove this?

File: python/lsst/ts/tcpip/testone_client_read_loop_server.py [hunk 2]
```diff
@@ -21,25 +21,61 @@
...
 
 class TestOneClientReadLoopServer(OneClientReadLoopServer):
-    """An implementation of OneClientReadLoopServer for unit test
-    purposes.
+    """A simple implementation of OneClientReadLoopServer for unit tests.
+
+    Parameters
+    ----------
+    host : `str` or `None`
+        IP address for this server; typically `LOCALHOST_IPV4` for IPV4
```
Comment (wvreeven) [inline python/lsst/ts/tcpip/testone_client_read_loop_server.py:38]: Would you like to mention the new `LOCALHOST` constant here?

File: python/lsst/ts/tcpip/base_one_client_server_test_case.py [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+    @contextlib.asynccontextmanager
+    async def create_server(
+        self,
+        host: str = LOCALHOST,
+        port: int = 0,
+        connect_callback: ConnectCallbackType | None | Literal[True] = True,
+        name: str = "",
+        **kwargs: Any,
+    ) -> AsyncGenerator[OneClientServer, None]:
+        """Create a server of the class being tested, and set as server.
```
Comment (wvreeven) [inline python/lsst/ts/tcpip/base_one_client_server_test_case.py:106]: Where is it set as server? The instance is returned, isn't it?

**Review by wvreeven — APPROVED** [review]
Looks good to me. I am looking forward to using this! Please see the few comments I left.

File: python/lsst/ts/tcpip/constants.py (hunk 1 above)
Comment (r-owen) [reply python/lsst/ts/tcpip/constants.py:40]: Good question. Perhaps for version 2? The problem is that I don't know how to make use raise a warning.

File: python/lsst/ts/tcpip/testone_client_read_loop_server.py (hunk 2 above)
Comment (r-owen) [reply python/lsst/ts/tcpip/testone_client_read_loop_server.py]: Good suggestion. Done for all server classes.

**Pull request merged**